application/src/main/java/org/togetherjava/tjbot/commands/moderation/AuditCommand.java

```diff
@@ -85,30 +90,24 @@ public AuditCommand(@NotNull ModerationActionsStore actionsStore, @NotNull Confi
         return shortSummary + "\n" + typeCountSummary;
     }
 
-    private static @NotNull RestAction<MessageEmbed> actionToMessage(@NotNull ActionRecord action,
+    private static @NotNull MessageEmbed.Field actionToField(@NotNull ActionRecord action,
             @NotNull JDA jda) {
-        String footer = action.actionExpiresAt() == null ? null
-                : "Temporary action, expires at %s".formatted(TimeUtil
-                    .getDateTimeString(action.actionExpiresAt().atOffset(ZoneOffset.UTC)));
-
-        return jda.retrieveUserById(action.authorId())
-            .onErrorMap(error -> null)
-            .map(author -> new EmbedBuilder().setTitle(action.actionType().name())
-                .setAuthor(author == null ? "(unknown user)" : author.getAsTag(), null,
-                        author == null ? null : author.getAvatarUrl())
-                .setDescription(action.reason())
-                .setTimestamp(action.issuedAt())
-                .setFooter(footer)
-                .setColor(ModerationUtils.AMBIENT_COLOR)
-                .build());
-    }
-
-    private static <E> @NotNull List<E> prependElement(@NotNull E element,
-            @NotNull Collection<? extends E> elements) {
-        List<E> allElements = new ArrayList<>(elements.size() + 1);
-        allElements.add(element);
-        allElements.addAll(elements);
-        return allElements;
+        Function<Instant, String> formatTime = instant -> {
+            if (instant == null) {
+                return "";
+            }
+            return TimeUtil.getDateTimeString(instant.atOffset(ZoneOffset.UTC));
+        };
+
+        User author = jda.getUserById(action.authorId());
+
+        return new MessageEmbed.Field(
+                action.actionType().name() + " by "
+                        + (author == null ? "(unknown user)" : author.getAsTag()),
+                action.reason() + "\nIssued at: " + formatTime.apply(action.issuedAt())
+                        + "\nTemporary action, Expires at: "
+                        + formatTime.apply(action.actionExpiresAt()),
```

Comment:
What if it is not temporary? This string should be missing entirely then. It looks to me as if it is still there now, no? As in `Temporary action, Expires at: "` and then no text anymore.

Btw. it should be `expires` not `Expires`.

Reply:
fixed